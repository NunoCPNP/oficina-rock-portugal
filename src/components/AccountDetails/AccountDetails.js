import useAuth from '@/hooks/useAuth'
import useTranslation from 'next-translate/useTranslation'

import { Block, Container, GridContainer, Info, Title } from './AccountDetails.styles'

const AccountDetails = () => {
  const { t } = useTranslation('common')
  const { user } = useAuth()

  if (!user) return null

  return (
    <Container>
      <GridContainer>
        <Block>
          <Title>
            <h2>{t(`personal-details`)}</h2>
          </Title>
          <Info>
            <span>{t(`name`)} : </span>
            {user.displayName}
          </Info>
          <Info>
            <span>{t(`email`)} : </span>
            {user.email}
          </Info>
          <Info>
            <span>{t(`phone-number`)} : </span>
            {user.phoneNumber}
          </Info>
        </Block>
        <Block>
          <Title>
            <h2>{t(`delivery-address`)}</h2>
          </Title>
          <Info>
            <span>{t(`street`)} : </span>
            {user.deliveryAddress.street}
          </Info>
          <Info>
            <span>{t(`postal-code`)} : </span>
            {user.deliveryAddress.postalCode}
          </Info>
          <Info>
            <span>{t(`city`)} : </span>
            {user.deliveryAddress.city}
          </Info>
          <Info>
            <span>{t(`country`)} : </span>
            {user.deliveryAddress.country}
          </Info>
        </Block>
        {user.deliverySameAsBilling ? (
          <Block>
            <div>{t(`same-address`)}</div>
          </Block>
        ) : (
          <Block>
            <Title>
              <h2>{t(`billing-address`)}</h2>
            </Title>
            <Info>
              <span>{t(`street`)} : </span>
              {user.deliveryAddress.street}
            </Info>
            <Info>
              <span>{t(`postal-code`)} : </span>
              {user.deliveryAddress.postalCode}
            </Info>
            <Info>
              <span>{t(`city`)} : </span>
              {user.deliveryAddress.city}
            </Info>
            <Info>
              <span>{t(`country`)} : </span>
              {user.deliveryAddress.country}
            </Info>
          </Block>
        )}
      </GridContainer>
    </Container>
  )
}

export default AccountDetails
