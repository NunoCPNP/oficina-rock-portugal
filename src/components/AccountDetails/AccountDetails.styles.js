import mq from '@/styles/mediaQueries'
import styled from '@emotion/styled'

export const Container = styled.div`
  margin: 4rem;

  ${mq('s')} {
    margin: 4rem 1rem;
  }
`

export const GridContainer = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;

  ${mq('s')} {
    grid-template-columns: 1fr;
  }
`

export const Title = styled.div`
  margin-bottom: 3rem;

  h2 {
    font-size: var(--heading-3);
  }
`

export const Info = styled.div`
  text-transform: uppercase;
  margin-bottom: 1rem;

  span {
    font-weight: 700;
    padding-right: 0.6rem;
  }
`

export const Block = styled.div`
  padding-bottom: 4rem;
`
