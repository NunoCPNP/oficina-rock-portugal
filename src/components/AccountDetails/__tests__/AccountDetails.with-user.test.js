import { render } from '@testing-library/react'

import AccountDetails from '../AccountDetails'

jest.mock('@/hooks/useAuth', () => {
  return jest.fn(() => ({
    user: {
      displayName: 'Semple User',
      deliveryAddress: {
        city: 'Matosinhos',
        country: 'Portugal',
        postalCode: '4450-120',
        street: 'Av Comendador Ferreira de Matos, 312, 2ยบ Esq',
      },
      deliverySameAsBilling: true,
      displayName: 'Nuno de Carvalho Pinheiro Nunes Pereira',
      email: 'hello@nunopereira.tech',
      phoneNumber: '968143844',
    },
  }))
})

describe('<AccountDetails />', () => {
  it('Should match snapshot', () => {
    const { container } = render(<AccountDetails />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
