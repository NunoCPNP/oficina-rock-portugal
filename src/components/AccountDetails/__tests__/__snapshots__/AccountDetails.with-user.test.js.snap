// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountDetails /> Should match snapshot 1`] = `
.emotion-0 {
  margin: 4rem;
}

@media only screen and (max-width: 480px) {
  .emotion-0 {
    margin: 4rem 1rem;
  }
}

.emotion-1 {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
}

@media only screen and (max-width: 480px) {
  .emotion-1 {
    grid-template-columns: 1fr;
  }
}

.emotion-2 {
  padding-bottom: 4rem;
}

.emotion-3 {
  margin-bottom: 3rem;
}

.emotion-3 h2 {
  font-size: var(--heading-3);
}

.emotion-4 {
  text-transform: uppercase;
  margin-bottom: 1rem;
}

.emotion-4 span {
  font-weight: 700;
  padding-right: 0.6rem;
}

<div
  class="emotion-0"
>
  <div
    class="emotion-1"
  >
    <div
      class="emotion-2"
    >
      <div
        class="emotion-3"
      >
        <h2>
          Dados Pessoais
        </h2>
      </div>
      <div
        class="emotion-4"
      >
        <span>
          Nome : 
        </span>
        Nuno de Carvalho Pinheiro Nunes Pereira
      </div>
      <div
        class="emotion-4"
      >
        <span>
          E-Mail : 
        </span>
        hello@nunopereira.tech
      </div>
      <div
        class="emotion-4"
      >
        <span>
          Telefone : 
        </span>
        968143844
      </div>
    </div>
    <div
      class="emotion-2"
    >
      <div
        class="emotion-3"
      >
        <h2>
          Morada de Envio
        </h2>
      </div>
      <div
        class="emotion-4"
      >
        <span>
          Rua : 
        </span>
        Av Comendador Ferreira de Matos, 312, 2º Esq
      </div>
      <div
        class="emotion-4"
      >
        <span>
          Código Postal : 
        </span>
        4450-120
      </div>
      <div
        class="emotion-4"
      >
        <span>
          Localidade : 
        </span>
        Matosinhos
      </div>
      <div
        class="emotion-4"
      >
        <span>
          País : 
        </span>
        Portugal
      </div>
    </div>
    <div
      class="emotion-2"
    >
      <div
        class="emotion-3"
      >
        <h2>
          Morada de Faturação
        </h2>
      </div>
      <div
        class="emotion-4"
      >
        <span>
          Rua : 
        </span>
        Av Comendador Ferreira de Matos, 312, 2º Esq
      </div>
      <div
        class="emotion-4"
      >
        <span>
          Código Postal : 
        </span>
        4450-120
      </div>
      <div
        class="emotion-4"
      >
        <span>
          Localidade : 
        </span>
        Matosinhos
      </div>
      <div
        class="emotion-4"
      >
        <span>
          País : 
        </span>
        Portugal
      </div>
    </div>
  </div>
</div>
`;
